RecoParticleFlow/PFRecHitProducer/plugins/alpaka/PFRecHitProducerKernel.h

```diff
@@ -0,0 +1,35 @@
+#ifndef RecoParticleFlow_PFRecHitProducer_plugins_alpaka_PFRecHitProducerKernel_h
+#define RecoParticleFlow_PFRecHitProducer_plugins_alpaka_PFRecHitProducerKernel_h
+
+#include "DataFormats/ParticleFlowReco/interface/alpaka/PFRecHitDeviceCollection.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "CalorimeterDefinitions.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  template <typename CAL>
+  class PFRecHitProducerKernel {
+  public:
+    PFRecHitProducerKernel(Queue& queue, const uint32_t num_recHits);
+
+    // Run kernel: apply filters to rec hits and construct PF rec hits
+    // This may be executed multiple times per event
+    void processRecHits(Queue& queue,
```

Comment:
_In principle_ given the way this class is used by the caller, this class could store a reference to the `Queue` as member data to avoid repeating in these functions calls.